### PR TITLE
`SemanticsFlag`/`SemanticsAction` cleanup (part 4)

### PR DESCRIPTION
**Pull request by bernaferrari**
part 4 of https://github.com/flutter/flutter/issues/123346

### CODE REVIEW

**Comment by goderbauer** [comment]
What about the matcher_test.dart in flutter_test that iterates over `SemanticsAction.values.keys`?

**Comment by bernaferrari** [comment]
That's an excellente question. I completely overlooked it. Thanks!

We could:
- Expose `_kActionById`, make it public (on AndroidSemanticsAction it is private, it is where I got the idea);
- Add a `@protected Iterable<int> keys => _kActionById.keys;`, since that's used in tests only.

What do you think?

**Comment by bernaferrari** [comment]
Oh, alternatively we could just do `SemanticsAction.values.map((d) => d.value))` right? Would work the same.

**Comment by goderbauer** [comment]
`SemanticsAction.values.map((d) => d.value))` sounds like the best option.

File: packages/flutter_test/lib/src/matchers.dart [hunk 1]
```diff
@@ -2533,7 +2533,7 @@ class _MatchesSemanticsData extends Matcher {
 
   static String _createEnumsSummary<T extends Object>(List<T> enums) {
     assert(T == SemanticsAction || T == SemanticsFlag, 'This method is only intended for lists of SemanticsActions or SemanticsFlags.');
-    return '[${enums.map(describeEnum).join(', ')}]';
+    return '[${enums.map((Object d) => d is SemanticsAction ? d.name : (d as SemanticsFlag).name).join(', ')}]';
```
Comment (bernaferrari) [inline packages/flutter_test/lib/src/matchers.dart:2536]: The only "not-exactly-expected" thing from this PR is this. Everything else is super-simple and super-expected.

**Comment by bernaferrari** [comment]
Cc @chunhtai

File: packages/flutter_test/test/matchers_test.dart [hunk 2]
```diff
@@ -613,12 +613,10 @@ void main() {
       int actions = 0;
       int flags = 0;
       const CustomSemanticsAction action = CustomSemanticsAction(label: 'test');
-      // ignore: deprecated_member_use
-      for (final int index in SemanticsAction.doNotUseWillBeDeletedWithoutWarningKeys) {
+      for (final int index in SemanticsAction.values.map((SemanticsAction value) => value.index)) {
```
Comment (goderbauer) [inline packages/flutter_test/test/matchers_test.dart:616]: Could this just be simplified to:

```dart
      for (final SemanticsAction action in SemanticsAction.values) {
        actions |= action.index;
      }
```

Comment (goderbauer) [reply packages/flutter_test/test/matchers_test.dart:616]: (same for all the cases below)

Comment (bernaferrari) [reply packages/flutter_test/test/matchers_test.dart:616]: Yesssss

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by goderbauer** [comment]
FYI, Google testing is currently failing because the engine-side change has not rolled into google yet.

**Review by chunhtai — APPROVED** [review]
LGTM

**Comment by bernaferrari** [comment]
I'll use my new super-power to auto-submit (I hope it is fine).

**Pull request merged**